app/src/features/editor/hooks/useCompile.tsx

```diff
@@ -1,25 +1,48 @@
 import styled from '@emotion/styled';
 import ansicolor from 'ansicolor';
 import React, { useState, useEffect } from 'react';
+import { saveAbi, saveBytecode } from '../../../utils/localStorage';
+import { CopyableHex } from '../../../components/shared';
+
+function toResults(
+  prefixedBytecode: string,
+  abi: string
+): React.ReactElement[] {
+  return [
+    <div key={'bytecode'}>
+      <b>Bytecode</b>:<br />
+      <CopyableHex hex={prefixedBytecode} />
+      <br />
+      <br />
+    </div>,
+    <div key={'abi'}>
+      <b>ABI:</b>
+      <br />
+      {abi}
+    </div>,
+  ];
+}
 
 export function useCompile(
   code: string | undefined,
-  onError: (error: string | undefined) => void
-): React.ReactElement[] {
-  const [results, setResults] = useState<React.ReactElement[]>([]);
+  onError: (error: string | undefined) => void,
+  setIsCompiled: (isCompiled: boolean) => void,
+  setResults: (entry: React.ReactElement[]) => void
+) {
   const [serverError, setServerError] = useState<boolean>(false);
 
   useEffect(() => {
     if (!code) {
       setResults([<>Click 'Compile' to build your code.</>]);
       return;
     }
-
-    if (!code.length) {
+    if (!code?.length) {
       setResults([<>Add some code to compile.</>]);
       return;
     }
 
+    setResults([<>Compiling...</>]);
+
     // TODO: Determine the URL based on the NODE_ENV.
     const server_uri = 'https://api.sway-playground.org/compile';
     // const server_uri = 'https://127.0.0.1/compile';
```

Comment:
Should we keep this line seems like left from debugging

Reply:
This prints `Compiling...` in the console that the user sees, so it's useful to keep it in :)

Reply:
Oh I meant line 48, but nothing important 🙌